app/components-react/root/NotificationsArea.tsx

```diff
@@ -1,58 +1,116 @@
 import React, { useState, useEffect, useRef } from 'react';
-import { Badge, Tooltip } from 'antd';
+import { injectState, useModule } from 'slap';
+import { Badge, message, Tooltip } from 'antd';
 import moment from 'moment';
 import cx from 'classnames';
-import { useVuex, useRenderInterval } from '../hooks';
 import cloneDeep from 'lodash/cloneDeep';
 import { Services } from '../service-provider';
 import { ENotificationType, INotification, ENotificationSubType } from 'services/notifications';
 import { $t } from 'services/i18n';
 import styles from './NotificationsArea.m.less';
 const notificationAudio = require('../../../media/sound/ding.wav');
 
-interface IUiNotification extends INotification {
-  outdated?: boolean;
+class NotificationsModule {
+  currentNotif: INotification | null = null;
+  notifQueue = cloneDeep(Services.NotificationsService.state.notifications);
+
+  audio = new Audio(notificationAudio);
+
+  // Add Notification to queue if one is playing, otherwise play it
+  addNotif(notif: INotification) {
+    if (this.currentNotif) {
+      this.notifQueue.push(notif);
+    } else {
+      this.playNotif(notif);
+    }
+  }
+
+  playNotif(notif: INotification) {
+    if (!this.settings.enabled) return;
+    this.currentNotif = notif;
+    if (this.settings.playSound) this.audio.play();
+
+    message
+      .info({
+        content: <MessageNode notif={notif} />,
+        duration: notif.lifeTime === -1 ? 0 : notif.lifeTime / 1000,
+        key: `${notif.message}${notif.date}`,
+        onClick: () => this.clickNotif(),
+        className: cx(styles.notification, {
+          [styles.info]: notif.type === 'INFO',
+          [styles.warning]: notif.type === 'WARNING',
+          [styles.success]: notif.type === 'SUCCESS',
+          [styles.hasAction]: notif.action,
+        }),
+      })
+      .then(() => this.playNext());
+  }
+
+  // Play next Notification in the queue, otherwise clear the current Notification
+  playNext() {
+    if (this.currentNotif) {
+      message.destroy();
```

Comment:
It seems weird to me that this is a global function rather than something we call on a specific message.

Reply:
i ran into some issues trying to destroy a specific message, but can try messing around with it again